src/website/app/demos/Menu/index.js

```diff
@@ -28,35 +28,65 @@ const menuDividerDoc = require('!!react-docgen-loader!../../../../Menu/MenuDivid
+Think through the consequenses of the actions in your Menu.
+If there are potentially destructive actions exposed, be sure to use the correct [variants](/color#color-variants) to provide a hint to the user about what will happen after picking an option.
+
+Labels should have clear messaging in the form of \`<verb> <noun>\` if possible.`
   },
   {
+    bestPractices: bestPractices.menuDivider,
     componentTheme: menuDividerComponentTheme,
     doc: menuDividerDoc,
     examples: menuDividerExamples,
     slug: 'menu-divider',
     subcomponent: true,
-    title: 'MenuDivider'
+    title: 'MenuDivider',
+    whenHowToUse: `MenuDividers are used to visually separate [MenuGroups](../menu-group) or individual [MenuItems](../menu-item) to establish hierarchy in your Menu.
+They could be used to separate items in a long list if all elements exist on the same conceptual level.
+For example, if your Menu is composed of a long list of songs, a MenuDivider could be placed between songs that start with A, B, C, etc.
+
+Do not use MenuDividers to simply separate options as a decoration, or to provide "underlines" to the MenuItem.`
   },
   {
+    bestPractices: bestPractices.menuGroup,
     componentTheme: menuGroupComponentTheme,
     doc: menuGroupDoc,
     examples: menuGroupExamples,
     slug: 'menu-group',
     subcomponent: true,
-    title: 'MenuGroup'
+    title: 'MenuGroup',
+    whenHowToUse: `MenuGroups are used to group conceptually related elements, and to hint at other available, related options.
+If the intention of the grouping is not immediately obvious, add a section title via the \`title\` attribute to aid in your users' decision-making process.
+
+If there are only a few elements in a couple of groups, and the grouping logic is obvious, consider using a [MenuDivder](../menu-divider) instead.`
   },
   {
+    bestPractices: bestPractices.menuItem,
     componentTheme: menuItemComponentTheme,
     doc: menuItemDoc,
     examples: menuItemExamples,
     slug: 'menu-item',
     subcomponent: true,
-    title: 'MenuItem'
+    title: 'MenuItem',
+    whenHowToUse: `Use MenuItems to present the user with a choice of actions.
+This may sound obvious, but don't use MenuItems to display content that is not actionable.
+For example, you might receive a spec for a [Menu](../menu) with several options where the last item is an un-clickable status message showing the number of available servers.
+This information is not _actionable_ and perhaps belongs somewhere else in your interface.
+
+Don't use \`disabled\` MenuItems as section titles or to create groups, instead set the \`title\` attribute on a [MenuGroup](../menu-group).
+
+Use the secondary text to give hints about extra functionality or provide status.
+
+If there are potentially destructive actions exposed, be sure to use the correct [variants](/color#color-variants) to provide a hint to the user about what will happen after picking an option.`
```

Comment:
We shouldn't use this space to enumerate each of the practices below. Rather, we should just call out the really key guidelines, as briefly as we can.